packages/server/graphql/mutations/joinMeeting.ts

```diff
@@ -0,0 +1,149 @@
+import {GraphQLID, GraphQLNonNull} from 'graphql'
+import {SubscriptionChannel} from 'parabol-client/types/constEnums'
+import {MeetingTypeEnum, NewMeetingPhaseTypeEnum} from '../../../client/types/graphql'
+import toTeamMemberId from '../../../client/utils/relay/toTeamMemberId'
+import getRethink from '../../database/rethinkDriver'
+import rMapIf from '../../database/rMapIf'
+import ActionMeetingMember from '../../database/types/ActionMeetingMember'
+import CheckInPhase from '../../database/types/CheckInPhase'
+import CheckInStage from '../../database/types/CheckInStage'
+import Meeting from '../../database/types/Meeting'
+import MeetingRetrospective from '../../database/types/MeetingRetrospective'
+import PokerMeetingMember from '../../database/types/PokerMeetingMember'
+import RetroMeetingMember from '../../database/types/RetroMeetingMember'
+import UpdatesPhase from '../../database/types/UpdatesPhase'
+import UpdatesStage from '../../database/types/UpdatesStage'
+import {getUserId, isTeamMember} from '../../utils/authorization'
+import publish from '../../utils/publish'
+import {GQLContext} from '../graphql'
+import JoinMeetingPayload from '../types/JoinMeetingPayload'
+
+const createMeetingMember = (meeting: Meeting, teamId: string, userId: string) => {
+  switch (meeting.meetingType) {
+    case MeetingTypeEnum.action:
+      return new ActionMeetingMember({teamId, userId, meetingId: meeting.id})
+    case MeetingTypeEnum.retrospective:
+      const {id: meetingId, totalVotes} = meeting as MeetingRetrospective
+      return new RetroMeetingMember({
+        teamId,
+        userId,
+        meetingId,
+        votesRemaining: totalVotes
+      })
+    case MeetingTypeEnum.poker:
+      return new PokerMeetingMember({
+        teamId,
+        userId,
+        meetingId: meeting.id
+      })
+    default:
+      throw new Error('Invalid meeting type')
+  }
+}
+
+const joinMeeting = {
+  type: GraphQLNonNull(JoinMeetingPayload),
+  description: `Create a meeting member for a user`,
+  args: {
+    meetingId: {
+      type: GraphQLNonNull(GraphQLID)
+    }
+  },
+  resolve: async (
+    _source,
+    {meetingId},
+    {authToken, dataLoader, socketId: mutatorId}: GQLContext
+  ) => {
+    const r = await getRethink()
+    const viewerId = getUserId(authToken)
+    const operationId = dataLoader.share()
+    const subOptions = {mutatorId, operationId}
+    //AUTH
+    const meeting = await dataLoader.get('newMeetings').load(meetingId)
+    if (!meeting) {
+      return {error: {message: 'Invalid meeting ID'}}
+    }
+    const {endedAt, teamId, phases} = meeting
+    if (endedAt) {
+      return {error: {message: 'Meeting has already ended'}}
+    }
+    if (!isTeamMember(authToken, teamId)) {
+      return {error: {message: 'Not on the team'}}
+    }
+    const meetingMember = createMeetingMember(meeting, teamId, viewerId)
+    const {errors} = await r
```

Comment:
since the id is a composite of the userId & meetingId, calling this concurrently will result in only 1 meeting member & failing on the 2nd call